pkg/pillar/cmd/zedagent/handleProfile.go

```diff
@@ -0,0 +1,326 @@
+	}
+
+	var errList []string
+	for bridgeName, urlAndSrcIPs := range localServerMap {
+		for _, el := range urlAndSrcIPs {
+			resp, contents, err := zedcloud.SendLocal(zedcloudCtx, el.actualURL, bridgeName, el.srcIP, 0, nil)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("SendLocal: %s", err))
+				continue
+			}
+			if resp.StatusCode != http.StatusOK {
+				errList = append(errList, fmt.Sprintf("SendLocal: wrong response status code: %d",
+					resp.StatusCode))
+				continue
+			}
+			if err := validateProtoMessage(el.actualURL, resp); err != nil {
+				// send something to ledmanager ???
+				errList = append(errList, fmt.Sprintf("validateProtoMessage: resp header error: %s", err))
+				continue
+			}
+			localProfile, err := parseAndValidateLocalProfile(contents, getconfigCtx)
+			if err != nil {
+				errList = append(errList, fmt.Sprintf("parseAndValidateLocalProfile: %s", err))
+				continue
+			}
+			return localProfile, nil
+		}
+	}
+	return nil, fmt.Errorf("getLocalProfileConfig: all attempts failed: %s", strings.Join(errList, ";"))
+}
+
+func writeReceivedProtoMessageLocalProfile(localProfile *profile.LocalProfile) {
+	contents, err := proto.Marshal(localProfile)
+	if err != nil {
+		log.Errorf("writeReceivedProtoMessageLocalProfile Marshalling failed: %s", err)
+		return
+	}
+	writeProtoMessage(savedLocalProfileFile, contents)
+	return
+}
+
+//parseProfile process local and global profile configuration
+//must be called before processing of app instances from config
+func parseProfile(ctx *getconfigContext, config *zconfig.EdgeDevConfig) {
+	log.Functionf("parseProfile start: globalProfile: %s localProfile: %s",
+		ctx.globalProfile, ctx.localProfile)
+	if ctx.globalProfile != config.GlobalProfile {
+		log.Noticef("parseProfile: GlobalProfile changed from %s to %s",
+			ctx.globalProfile, config.GlobalProfile)
+		ctx.globalProfile = config.GlobalProfile
+	}
+	if ctx.localProfileServer != config.LocalProfileServer {
+		// read saved config on the first call
+		if !ctx.callProcessProfile {
+			if localProfile, err := readSavedLocalProfile(ctx, false); err != nil {
+				log.Functionf("parseProfile readSavedLocalProfile: %s", err)
+			} else if localProfile != nil {
+				log.Noticef("starting with localProfile %s", localProfile.LocalProfile)
+				ctx.localProfile = localProfile.LocalProfile
+			}
+			ctx.callProcessProfile = true
+		} else {
+			// if localProfileServer changed not in the first call of function
+			// clean saved proto message and set localProfile to empty
+			cleanSavedProtoMessage(savedLocalProfileFile)
+			ctx.localProfile = ""
+		}
+		log.Noticef("parseProfile: LocalProfileServer changed from %s to %s",
+			ctx.localProfileServer, config.LocalProfileServer)
+		ctx.localProfileServer = config.LocalProfileServer
+		if ctx.localProfileServer != "" {
+			// notify to reload local profile on profileServerChange
+			select {
+			case ctx.localProfileTrigger <- Notify{}:
+			default:
+			}
```

Comment:
Define a function for this, since you'll be calling it in more than one place? Or have a local notify boolean to determine whether to do this.